packages/gatsby-graphiql-explorer/src/app/app.js

```diff
@@ -0,0 +1,159 @@
+import React from "react"
+import ReactDOM from "react-dom"
+
+import GraphiQL from "graphiql"
+import GraphiQLExplorer from "graphiql-explorer"
+import { getIntrospectionQuery, buildClientSchema } from "graphql"
+
+import "whatwg-fetch"
+
+import "graphiql/graphiql.css"
+import "./app.css"
+
+const parameters = {}
+window.location.search
+  .substr(1)
+  .split(`&`)
+  .forEach(function(entry) {
+    var eq = entry.indexOf(`=`)
+    if (eq >= 0) {
+      parameters[decodeURIComponent(entry.slice(0, eq))] = decodeURIComponent(
+        entry.slice(eq + 1)
+      )
+    }
+  })
+// Produce a Location query string from a parameter object.
+function locationQuery(params) {
+  return (
+    `?` +
+    Object.keys(params)
+      .filter(function(key) {
+        return Boolean(params[key])
+      })
+      .map(function(key) {
+        return encodeURIComponent(key) + `=` + encodeURIComponent(params[key])
+      })
+      .join(`&`)
+  )
+}
+
+// Derive a fetch URL from the current URL, sans the GraphQL parameters.
+const graphqlParamNames = {
+  query: true,
+  variables: true,
+  operationName: true,
+}
+const otherParams = {}
+for (var k in parameters) {
+  if (parameters.hasOwnProperty(k) && graphqlParamNames[k] !== true) {
+    otherParams[k] = parameters[k]
+  }
+}
+const fetchURL = locationQuery(otherParams)
+
+function graphQLFetcher(graphQLParams) {
+  return fetch(fetchURL, {
+    method: `post`,
+    headers: {
+      Accept: `application/json`,
+      "Content-Type": `application/json`,
+    },
+    body: JSON.stringify(graphQLParams),
+    credentials: `include`,
+  }).then(function(response) {
+    return response.json()
+  })
+}
+
+// When the query and variables string is edited, update the URL bar so
+// that it can be easily shared.
+function onEditVariables(newVariables) {
+  parameters.variables = newVariables
+  updateURL()
+}
+function onEditOperationName(newOperationName) {
+  parameters.operationName = newOperationName
+  updateURL()
+}
+function updateURL() {
+  history.replaceState(null, null, locationQuery(parameters))
+}
+
+// We control query, so we need to recreate initial query text that show up
+// on visiting graphiql - in order it will be
+//  - query from query string (if set)
+//  - query stored in localStorage (which graphiql set when closing window)
+//  - default empty query
+const DEFAULT_QUERY =
+  parameters.query ||
+  (window.localStorage && window.localStorage.getItem(`graphiql:query`)) ||
+  `{
+
+}`
```

Comment:
here we could add comments that will guide first time users

Reply:
I should just copy & paste default graphiql "query" ( https://github.com/graphql/graphiql/blob/25006623dc56c78259cd1ab1401eea6b96343b42/src/components/GraphiQL.js#L1027-L1058 )

Reply:
This works with me. We could customize the query to be Gatsby specific too, e.g. a query that will typically always work, a la

```graphql
{
  site {
    siteMetadata {
      title
    }
  }
}
```

(also open to anything else here--just enough to convey a simple query that will rarely, ideally never, break)

Reply:
This sounds good. 

I'll check how hard is to see if `site.siteMetadata.title` is actually in schema (we do instrospection query, so this information should be available) and will just fallback to `allSitePage.nodes.path` which will be available always (but not as useful as siteMetadata query)

Reply:
Done